Tasks/DotNetCoreCLIV2/dotnetcore.ts

```diff
@@ -325,22 +326,50 @@ export class dotNetExe {
         }
 
         var projectFiles = utility.getProjectFiles(projectPattern);
+        var resolvedProjectFiles: string[] = [];
 
         if (searchWebProjects) {
-            projectFiles = projectFiles.filter(function (file, index, files): boolean {
+            resolvedProjectFiles = projectFiles.filter(function (file, index, files): boolean {
                 var directory = path.dirname(file);
                 return tl.exist(path.join(directory, "web.config"))
                     || tl.exist(path.join(directory, "wwwroot"));
             });
 
-            if (!projectFiles.length) {
-                tl.error(tl.loc("noWebProjectFound"));
-            }
+            if (!resolvedProjectFiles.length) {
+                var projectFilesUsingWebSdk = projectFiles.filter(this.isWebSdkUsed)
+                if(!projectFilesUsingWebSdk.length) {
+                    tl.error(tl.loc("noWebProjectFound"));
+                }
+                return projectFilesUsingWebSdk;
+            } 
+            return resolvedProjectFiles;
         }
-
         return projectFiles;
     }
 
+    private isWebSdkUsed(projectfile: string): boolean {
+        try {
+            var fileBuffer: Buffer = fs.readFileSync(projectfile);
+            var webConfigContent: string = fileBuffer.toString();
+            
+            var fileEncodings = ['utf8', 'utf16le', 'ascii', 'base64', 'binary', 'hex']
```

Comment:
I dont think we should only expect utf8 or utf16le. This is what we support in https://github.com/microsoft/azure-pipelines-tasks/blob/master/Tasks/AzureResourceGroupDeploymentV2/operations/FileEncoding.ts here well

Reply:
These are the encodings supported by node js's file buffer. if the file is of different encoding, we'd need to find a way to parse the encoding itself..

Reply:
fileEncoding.ts tries to do that. However, i am fine with your approach as well. However, what I am saying is that you can remove everything except first 2 entries. We need not support other encodings.

Reply:
Got it, I've made the changes